lib/src/domain/model/article.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'article.g.dart';
+
+@JsonSerializable()
+class Article {
+  final int id;
+
+  final String title;
+
+  Article({this.id, this.title});
+
+  factory Article.fromJson(Map<String, dynamic> json) => _$ArticleFromJson(json);
```

Comment:
jeżeli klasa article to domain to powinna być to czysto dartowa klasa, która nie powinna wiedzieć jak wygląda deserializacja, która sama w sobie powinna mieć miejsce w warstwie data

Reply:
racja, tym bardziej, że będzie sporo pośrednich obiektów zrobionych tylko po to, żeby zagnieżdżone elementy jsona wyciągać